### ISSUE TITLE
Is sigmoid classifier suitable for multi-classification(num of categories > 1000 in LVIS) problems？

### ISSUE THREAD

**Issue by Icecream-blue-sky** [ModelTC/United-Perception#26]
Since you are based on the sigmoid classifier, I am curious if your detection results on LVIS will have many false positives in the same location but with different categories. The reason why I ask this is that, I used to train one-stage detector on datasets similar with LVIS (which is long tailed logo dataset, with 352 categories), however, I get many FP with different categories at the same location. I'm wondering if you have encountered the same situation. Thanks!
![alfaromeo5](https://user-images.githubusercontent.com/26771037/177005030-2828077e-8fa9-4f8e-939c-d0fb6d40e4a3.jpg)
And I think it may be due to the use of sigmoid classifier which consists of multiple independent binary classifiers. It may be not suitable for multi-classification(num of categories > 1000). Of course this is just my conjecture, any advice is welcome...

**Comment by waveboo** [comment]
Hi @Icecream-blue-sky , thanks for your good question~
Actually, the false positive problem is not specific to the sigmoid loss. Since the NMS in the modern detection framework is class-related, objects of different categories at the same location will not suppress each other. This problem is also in the softmax loss, although there is strong competition between different categories in the loss design. What's more, there are indeed many categories that are semantically similar. like mirrors and rear mirrors, dresses and costumes... I presume your task has a similar problem, because different logos may do not look much different.
To conquer this problem, on the one hand, you could apply the class-agnostic NMS to your detection result during the inference phase to filter out some low confidence boxes. And on the other hand, adding a larger penalty term for false positives during training is also a good option.

**Comment by Icecream-blue-sky** [comment]
> 

Thanks for your kind advice! Now I think part of the reason is the imbalance of training data(long tailed) in each category, which makes it difficult for rare classifiers to classify accurately. Maybe adding EFL loss would help, what do you think?
